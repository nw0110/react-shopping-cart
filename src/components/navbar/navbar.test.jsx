import {render} from '@testing-library/react';
import {describe, it, expect, vi} from 'vitest';

import Navbar from "./navbar.jsx";

vi.mock('react-router-dom', () => ({
    Link: ({ children, to }) => <a href={to}>{children}</a>
}));

describe('Navbar', () => {
    it('renders Navbar', () => {
        const { container } = render(<Navbar quantity={12} currentTotal={12}/>);
        expect(container).toMatchSnapshot();
    });
});