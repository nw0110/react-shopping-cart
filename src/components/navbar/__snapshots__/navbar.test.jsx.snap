// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders Navbar 1`] = `
<div>
  <div
    class="_navbar_8210b0"
  >
    <div
      class="_branding_8210b0"
    >
      <a
        href="/"
      >
        <img
          class="_mainLogo_8210b0"
          src="/src/assets/store.png"
        />
        <h1>
          SuperStore
        </h1>
      </a>
    </div>
    <nav
      class="_navElement_8210b0"
    >
      <a
        href="/cart"
      >
        <div
          class="_cartIconWithQuantity_8210b0"
        >
          <img
            height="32"
            src="/src/assets/cart.png"
          />
          <div
            class="_cartQuantity_8210b0"
          >
            12
          </div>
        </div>
        <div>
          12â‚¬
        </div>
      </a>
    </nav>
  </div>
</div>
`;
